packages/backend/src/index.ts

```diff
@@ -47,33 +46,28 @@ import graphql from './plugins/graphql';
 import app from './plugins/app';
 import { PluginEnvironment } from './types';
 
-function makeCreateEnv(loadedConfigs: AppConfig[]) {
-  const config = ConfigReader.fromConfigs(loadedConfigs);
+function makeCreateEnv(config: ConfigReader) {
+  const databaseManager = new SingleConnectionManager(
+    config.getConfig('backend.database'),
+  );
 
   return (plugin: string): PluginEnvironment => {
     const logger = getRootLogger().child({ type: 'plugin', plugin });
-    const database = createDatabaseClient(
-      config.getConfig('backend.database'),
-      {
-        connection: {
-          database: `backstage_plugin_${plugin}`,
-        },
-      },
-    );
+    const databaseFactory = databaseManager.getDatabaseClientFactory(plugin);
```

Comment:
Thought about this a bit more. Wondering if it might be a big cleaner to combine the concept of the manager and the factor into simply `DatabaseManager`. Essentially it could look something like this:

```ts
function makeCreateEnv(config: ConfigReader) {
  const database = SingleConnectionDatabaseManager.fromConfig(config);

  return (plugin: string): PluginEnvironment => {
    // do some stuff 
    return {
      logger,
      database, // NOTE: This could be renamed since it's not a factory by the manager (more succinct)
      config,
      discovery,
    };
  }
}
```

Reply:
I actually think that the current implementation is equivalent to this, just pending the other decision on whether to use an interface or to define a function type. This would really just boil down some renaming. Once again, it's preference and I'm agreeable either way.

Reply:
Addressed in https://github.com/spotify/backstage/pull/2697/commits/35c0294018b8433787a795b7d9fcb6e9daef2b68.